vllm/sequence.py

```diff
@@ -171,13 +171,13 @@ class SequenceData(msgspec.Struct,
     _mrope_position_delta: Optional[int] = None
 
     @staticmethod
-    def from_counts(counts_by_token: Mapping[int, int]) -> "SequenceData":
-        if len(counts_by_token) == 0:
+    def from_token_counts(*token_counts: Tuple[int, int]) -> "SequenceData":
+        if len(token_counts) == 0:
```

Comment:
I missed #8675 that refactored this, but imo it's probably a better to rename to something like `init_from_count_per_token_id`, and add WDYT?

Also, just so I understand, is the change to accommondate the scenario where one wants to precisely put where these tokens are? (e.g, to make a dummy sequence that looks like something like `[1, 1, 1, 0, 0, 0, 1, 1, 1]` instead of `[1, 1, 1, 1, 1, 1, 0, 0 ,0]` if we use mapping). If so, why/when we need to deal with this scenario?

If so, is there really a need for this?

Reply:
This change is to better handle the case when the token IDs from the HF config somehow duplicate other tokens passed to the constructor. Normally, this shouldn't happen, but the previous definition would silently drop those tokens instead of concatenating them which isn't the expected result.

Reply:
> it's probably better to rename it to something like `init_from_count_per_token_id`, and add a brief docstring for this function. WDYT?

I think `init_` is a bit redundant since the class name is already in front of the method name. The reason why I named it `from_token_counts` is to maintain the same order of `token` and `counts` as the tuple elements. I agree on the docstring though - since the CI for this PR seems to be passing already, are you ok with doing this in a subsequent PR?

Reply:
Yep that is fine!